piu/cli.py

```diff
@@ -168,36 +187,38 @@ def _request_access(even_url, cacert, username, hostname, reason, remote_host, l
     )
     if r.status_code == 200:
         click.secho(r.text, fg="green", bold=True)
-        ssh_command = ""
-        if remote_host:
-            ssh_command = "ssh -o StrictHostKeyChecking=no {username}@{remote_host}".format(**vars())
-            if tunnel:
-                ports = tunnel.split(":")
-                ssh_command = "-L {local_port}:{remote_host}:{remote_port}".format(
-                    local_port=ports[0], remote_host=remote_host, remote_port=ports[1]
-                )
-        command = "ssh -tA {username}@{hostname} {ssh_command}".format(
-            username=username, hostname=hostname, ssh_command=ssh_command
-        )
-        if connect or tunnel:
-            subprocess.call(command.split())
-
-        if not ssh_keys_added():
-            warning("No SSH identities found. Please add one using ssh-add, for example:")
-            warning("ssh-add ~/.ssh/id_rsa")
-
-        click.secho("You can access your server with the following command:")
-        click.secho(command)
-
-        if clip:
-            click.secho('\nOr just check your clipboard and run ctrl/command + v (requires package "xclip" on Linux)')
-            if pyperclip is not None:
-                pyperclip.copy(command)
+        ssh_connection(clip, connect, hostname, remote_host, tunnel, username, username)
     else:
         click.secho(
             "Server returned status {code}: {text}".format(code=r.status_code, text=r.text), fg="red", bold=True
         )
-    return r.status_code
+    return r.status_code == 200
+
+
+def ssh_connection(clip, connect, hostname, remote_host, tunnel, odd_user, remote_user):
+    if tunnel:
+        ports = tunnel.split(":")
+        ssh_command = "ssh {odd_user}@{hostname} -L {local_port}:{remote_host}:{remote_port}".format(
```

Comment:
I'm a bit unsure of scoping rules in python but this variable is not defined outside of the if. Is this a problem? :sweat_smile:

Reply:
It works fine this way.